src/compas/datastructures/assembly/part.py

```diff
@@ -0,0 +1,168 @@
+from __future__ import print_function
+from __future__ import absolute_import
+from __future__ import division
+
+from collections import deque
+from functools import reduce
+
+from compas.geometry import Shape
+from compas.geometry import Frame
+from compas.geometry import multiply_matrices
+from compas.geometry import boolean_union_mesh_mesh
+from compas.geometry import boolean_difference_mesh_mesh
+from compas.geometry import boolean_intersection_mesh_mesh
+
+from ..datastructure import Datastructure
+from ..mesh import Mesh
+
+from .exceptions import FrameError
+from .exceptions import FeatureError
+
+
+class Part(Datastructure):
+    """A data structure for representing assembly parts."""
+
+    operations = {
+        'union': boolean_union_mesh_mesh,
+        'difference': boolean_difference_mesh_mesh,
+        'intersection': boolean_intersection_mesh_mesh
+    }
+
+    @property
+    def DATASCHEMA(self):
+        import schema
+        return schema.Schema({
+            "attributes": dict,
+            "key": int,
+            "frame": Frame,
+            "transformations": list,
+            "geometry": Shape,
+            "features": list
+        })
+
+    @property
+    def JSONSCHEMANAME(self):
+        return 'assembly'
```

Comment:
I guess this one should be 

```suggestion
        return 'part'
```